### PR TITLE
Starting architecture documentation

### PR DESCRIPTION
**Pull request by danhorst**
Initial draft of our functional goals for CurateND. Some of these functions will be provided by ancillary systems not contained in this repo. In that sense the scope of the document is a bit broad. However, when we talk about “CurateND” to our stakeholders we are talking about _all_ of these systems as a whole as opposed to this particular Rails app.

### CODE REVIEW

File: architecture/Goals.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	- individuals to edit an existing work or file
+	- groups to view a work or file
+	- groups to edit an existing work or file
+- Allow members of the Notre Dame community to sponsor individuals:
+	- with no ND affiliation so they can participate in CurateND without a netID
+	- leaving Notre Dame so that they may continue to participate in CurateND without a netID
+- Allow individuals outside of the Notre Dame community:
+	- to view works and files
+	- to edit existing work and files
+	- to be members of a group
```
Comment (jeremyf) [inline architecture/Goals.md]:
```markdown
- Allow individuals to associate with other authentication systems:
  - A Notre Dame user could associate their CurateND account with ORCID (or GMail)
  - A sponsored user could associate with other identifiers (i.e. Gmail, ORCID, etc.)
```

**Comment by jeremyf** [comment]
This is an excellent framing document. I added a comment concerning authentication/identifier association.

File: architecture/Goals.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+### Versioning Content
+- Create separate works in CurateND for each version
```
Comment (jeremyf) [inline architecture/Goals.md]: What do you mean by this? A new PID for a new version?

**Comment by jeremyf** [comment]
Are we interested in exposing public APIs for read operations? What about write operations?

**Comment by danhorst** [comment]
I would like a :+1: / :-1: from @banurekha, @cdeluca495, @dbrower, @jeremyf, @rbalekai, and @rickjohnson to see if this captures our _current_ understanding of the expectations for CurateND.

The [markdown preview of the document](https://github.com/ndlib/curate_nd/blob/architecture/architecture/Goals.md) is a bit easier to read.

File: architecture/Goals.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+- Allow individuals to log in using other credentials than their netID:
+	- A Notre Dame user could associate their CurateND account with ORCID (or Gmail)
+	- A sponsored user could associate with other identifiers (i.e. Gmail, ORCID, etc.)
+- Allow sponsored individuals outside of the Notre Dame community:
+	- to view works and files
+	- to edit existing work and files
+	- to be members of a group
+
+### Versioning Content
+- Allow works to have multiple versions
```
Comment (jeremyf) [inline architecture/Goals.md]: I believe we need to vet the behavior of this with some user scenarios and policy.

What do we mean by version? Revision? When we change the metadata of a work, what are expectations about the previous state of the work? When we change a file how do we track revisions? When we make a successor version (i.e. a 2nd edition) what is the expectation? Do we associate with the 1st edition? How is that association tracked?

Do we need to expose editions and revisions?

This is something that I also believe should be role-played with some cards.

Comment (jeremyf) [inline architecture/Goals.md]: Should we be using language like "Make a new edition" or "Update this version"

**Comment by cdeluca495** [comment]
:+1:

**Comment by rbalekai** [comment]
:+1:

**Pull request merged**